source/common/router/config_impl.cc

```diff
@@ -79,6 +83,15 @@ bool RouteEntryImplBase::matches(const Http::HeaderMap& headers, uint64_t random
     matches &= (headers.get(Http::Headers::get().ContentType) == content_type_);
   }
 
+  if (!config_headers_.empty()) {
+    for (const Json::Object& header_map : config_headers_) {
```

Comment:
i'd also include contentType header here (and do not have separate content_type_) unless i'm missing something